### PR TITLE
Switch Translation Manager to RecyclerView

### LINKED ISSUE #709: Switch Translation Manager to RecyclerView

### PR DESCRIPTION
**Pull request by hosainnet**
Fixes #709 

- Introduced Butterknife for view injection
- Removed ListView and introduced RecyclerView
- Moved TranslationAdapter out of the TranslationActivity
- Used Rx subjects for handling download/remove events
- Used Snackbar instead of textview for error messages 

Any suggestions welcome

### CODE REVIEW

File: app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java [hunk 1]
```diff
@@ -222,23 +217,20 @@ private void generateListItems() {
...
-    items = res;
-    adapter.setData(items);
+    adapter.setTranslations(result);
     adapter.notifyDataSetChanged();
   }
 
-  private void downloadItem(int pos) {
-    if (items == null || !(adapter.getItem(pos) instanceof TranslationItem)) {
+  private void downloadItem(TranslationRowData translationRowData) {
+    if (!(translationRowData instanceof TranslationItem)) {
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java]: this `instanceof` is no longer needed

File: app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java [hunk 2]
```diff
@@ -173,7 +168,9 @@ private void updateTranslationItem(TranslationItem updated) {
   }
 
   public void onErrorDownloadTranslations() {
-    messageArea.setText(R.string.error_getting_translation_list);
+    Snackbar
```
Comment (hosainnet) [inline app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java]: 👍

File: app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java [hunk 3]
```diff
@@ -47,39 +43,36 @@
...
   private TranslationsAdapter adapter;
   private TranslationItem downloadingItem;
   private String databaseDirectory;
   private QuranSettings quranSettings;
-
-  @Inject TranslationManagerPresenter mPresenter;
   private DefaultDownloadReceiver mDownloadReceiver = null;
 
+  @Inject
+  TranslationManagerPresenter mPresenter;
```
Comment (hosainnet) [inline app/src/main/java/com/quran/labs/androidquran/ui/TranslationManagerActivity.java]: Sounds good, not a big fan of Hungarian notation anyway :)

File: app/src/main/java/com/quran/labs/androidquran/ui/adapter/TranslationsAdapter.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+
+import static android.text.TextUtils.isEmpty;
+
+public class TranslationsAdapter extends RecyclerView.Adapter<TranslationsAdapter.TranslationViewHolder> {
+
+  private final PublishSubject<TranslationRowData> onClickDownloadSubject = PublishSubject.create();
```
Comment (hosainnet) [inline app/src/main/java/com/quran/labs/androidquran/ui/adapter/TranslationsAdapter.java]: Ah didn't know about this, thanks!

**Comment by ahmedre** [comment]
@hosainnet jazakAllah khairan, i'll go ahead and merge this - in another patch, can you take care of moving the `onClickListener` to the `TranslationViewHolder`'s constructor?

also, can we unsubscribe to the 2 subjects in `onDestroy`?

**Pull request merged**